File: clases 25-11-2025/Tipos de clases js/clase normal con interfaz/Clases/Persona.js
import IPersona from '../interfaces/IPersona.js';

export default class Persona extends IPersona {
    constructor(nombre, edad) {
        super();
        this.nombre = nombre;
        this.edad = edad;
    }

    static from(otraPersona) {
        return new Persona(otraPersona.nombre, otraPersona.edad);
    }

    saludar() {
        console.log(`Hola, soy ${this.nombre} y tengo ${this.edad} años.`);
    }
}

File: clases 25-11-2025/Tipos de clases GO/clase_normal_con_interfaz/Persona.go
package main

import "fmt"

type IPersona interface {
	Saludar()
}

type Persona struct {
	nombre string
	edad   int
}

func NewPersona(nombre string, edad int) IPersona {
	return Persona{nombre: nombre, edad: edad}
}

func (p Persona) Saludar() {
	fmt.Printf("Hola, soy %s y tengo %d años.\n", p.nombre, p.edad)
}
